### ISSUE TITLE
Duplicate base URL

### ISSUE THREAD

**Issue by tkyi** [github-release/github-release#102]
We're running the github-release command like this: https://github.com/screwdriver-cd/toolbox/blob/master/git-release.sh#L38

With the latest release (v0.8.0), we're seeing failures due to a duplicate base URL in release command:
```
00:08:16 Creating release v1.0.510 for screwdriver-cd / ui
00:08:17 error: Get "https://api.github.comhttps//api.github.com/repositories/64884594/releases?per_page=100&page=2": dial tcp: lookup api.github.comhttps on 100.64.0.10:53: no such host
```

Could it be because there's already a defaultBaseUrl for NewClient?

**Comment by kevinburke** [comment]
Hi, thank you for the report, investigating now.

**Comment by kevinburke** [comment]
Someone filed an issue with a similar complaint recently: https://github.com/meterup/github-release/issues/1

However the script indicates you're downloading from the compiled version and that should have fixed the issue linked there. I also tried using the downloaded version and wasn't able to reproduce. I'm still investigating.

**Comment by tkyi** [comment]
Here's the full log if that helps: https://cd.screwdriver.cd/pipelines/7/builds/233174/steps/publish

**Comment by kevinburke** [comment]
Thanks, I'm attempting to reproduce using your script in a Docker container.

**Comment by kevinburke** [comment]
Here's my Dockerfile:

```
$ cat Dockerfile
FROM ubuntu:bionic
COPY . /go
WORKDIR /go
ARG GITHUB_TOKEN
ENV GITHUB_TOKEN=$GITHUB_TOKEN
ENV RELEASE_FILE="./git-release.sh"
ENV GIT_TAG="issue-102-testing"
ENV GIT_ORG=kevinburke
ENV GIT_REPO=github-release
RUN apt-get update && apt-get install -y wget git
RUN bash ./git-release.sh
```

Replace GIT_ORG and GIT_REPO with a repo that your token has write permission for. And bash script - I modified to hard code the GIT_ORG and GIT_REPO and print out all instructions as they run.

```
#!/bin/bash -e
# GITHUB_TOKEN = Token for updating your git repo
# RELEASE_FILE = File to release

set -euo pipefail
set -x

if [ -z "$GITHUB_TOKEN" ] || [ -z "$RELEASE_FILE" ]; then
  echo Unable to release, missing environment variables
  exit 2
fi

GITHUB_RELEASE=/tmp/linux-amd64-github-release
if [ ! -f "$GITHUB_RELEASE" ] ; then
  echo Downloading github-release
  wget -q -O - https://github.com/github-release/github-release/releases/latest \
     | egrep -o '/github-release/github-release/releases/download/v[0-9.]*/linux-amd64-github-release.bz2' \
     | wget --base=http://github.com/ -i - -O /tmp/linux-amd64-github-release.bz2
  bzip2 -dv /tmp/linux-amd64-github-release.bz2
  chmod +x $GITHUB_RELEASE
fi

GIT_VERSION=/tmp/gitversion
if [ ! -f "$GIT_VERSION" ] ; then
    echo Downloading gitversion
    wget -q -O - https://github.com/screwdriver-cd/gitversion/releases/latest \
       | egrep -o '/screwdriver-cd/gitversion/releases/download/v[0-9.]*/gitversion_linux_amd64' \
       | wget --base=http://github.com/ -i - -O /tmp/gitversion
    chmod +x $GIT_VERSION
fi

#GIT_ORG=`git remote -v | grep fetch | sed 's/ (fetch)//' | cut -d'/' -f4`
#GIT_REPO=`git remote -v | grep fetch | sed 's/ (fetch)//' | cut -d'/' -f5`
#if [ -f VERSION ] ; then
  #GIT_TAG=$(<VERSION)
#else
  #GIT_TAG=`$GIT_VERSION --prefix v show`
#fi

echo "Creating release $GIT_TAG for $GIT_ORG / $GIT_REPO"

$GITHUB_RELEASE release --user $GIT_ORG --repo $GIT_REPO --tag $GIT_TAG --name $GIT_TAG
$GITHUB_RELEASE upload --user $GIT_ORG --repo $GIT_REPO --tag $GIT_TAG --name $RELEASE_FILE --file $RELEASE_FILE
```

I successfully created the release and uploaded the file: https://github.com/kevinburke/github-release/releases/tag/issue-102-testing

Is it possible you have an older, or different, version of github-release on the machine that is doing the upload?

**Comment by kevinburke** [comment]
(Put Dockerfile and `git-release.sh` in the same directory then run 

```
docker build --build-arg GITHUB_TOKEN=$(cat ~/cfg/personal-github-token ) .
```

(get your GITHUB_TOKEN from wherever you get the token from)

**Comment by kevinburke** [comment]
I just pushed v0.8.1, which should provide more defense against a (possible) way that you ran into the issue above - if you could add `$(GITHUB_RELEASE) --version` to your script so I can ensure you're running the version I just released that would help with troubleshooting!

**Comment by tkyi** [comment]
Confirmed it has v0.8.1 and it published successfully! Thank you
https://cd.screwdriver.cd/pipelines/7/builds/238720/steps/publish

What changed?

**Comment by kevinburke** [comment]
Frankly the whole thing leaves me a little confused. The most likely culprit is github-release compiled an outdated version of a dependency, github.com/kevinburke/rest, which displayed exactly the error that you presented at the top of the ticket.

However, I patched the dependency a while ago, and included the patched release in both the 0.8.0 and 0.8.1 binaries. If you were downloading a 0.8.x binary, you should have gotten the newer library version without doing anything. Also, if the 0.8.0 binary included the error, I would have expected many more people to write in with complaints.

0.8.1 vendors every dependency, which was a way to triple verify that the updated library version was included in the binary.
